homeassistant/components/modbus/sensor.py

```diff
@@ -121,84 +121,97 @@ async def async_setup_platform(
             del entry[CONF_REGISTER_TYPE]
 
     for entry in discovery_info[CONF_SENSORS]:
+        if CONF_HUB in entry:
+            # from old config!
+            hub = hass.data[MODBUS_DOMAIN][entry[CONF_HUB]]
+        else:
+            hub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
+        if CONF_SCAN_INTERVAL not in entry:
+            entry[CONF_SCAN_INTERVAL] = DEFAULT_SCAN_INTERVAL
+        sensors.append(ModbusRegisterSensor(hub, entry))
+
+    async_add_entities(sensors)
+
+
+class ModbusRegisterSensor(RestoreEntity, SensorEntity):
+    """Modbus register sensor."""
+
+    @staticmethod
+    def _create_data_structure(entry: dict) -> tuple[str | None, str | None]:
```

Comment:
Will this be moved to a common base class in the future?

Reply:
not planned, this is only for platform sensor.

Reply:
Ok. Will there be another sensor class that can benefit from this? I'm wondering about the refactor reason.

Reply:
Jan did a good job doing the same for the Modbus Switch. With this PR, Modbus classes can be easily subclasses by the custom components (hacs). Not something super important, but good to have.

Reply:
I don't think we should refactor this integration just for custom integrations' sake. There should be a reason that benefits this integration, otherwise I suggest we close here.

Reply:
seen solely from modbus integration pow this PR is nice (because it unifies the platform classes) but in reality the integration have no benefit.

The only overall benefit is to have one more developer.